### PR TITLE
T&C

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/components/legal/legal-modal/legal-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export const LegalModalComponent = (
+    props: IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+) => {
+    const [showModal, setShowModal] = useState<boolean>(false);
+
+    useEffect(() => {
+        const tcAcceptedVersion = getTCVersion();
+
+        if (tcAcceptedVersion !== undefined) {
+            props.appSetAcceptedTcVersion(tcAcceptedVersion);
```
Comment (krisboit) [inline src/components/legal/legal-modal/legal-modal.tsx]: i don't think this is right ... if user didn't accept t&c before, it accepts latest version automatically. T&c should be flagged as accepted only when the user taps accept button

File: src/components/legal/legal.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { TC } from './terms-conditions/terms-conditions';
+import { PrivacyPolicy } from './privacy-policy/privacy-policy';
+import { IReduxState } from '../../redux/state';
+
+interface IExternalProps {
+    onAccept: () => void;
+}
+
+export interface IReduxProps {
+    tcVersion: number;
```
Comment (krisboit) [inline src/components/legal/legal.tsx]: why isn't this set from `mapStateToProps` ?

File: src/components/legal/legal.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                        <Icon name="chevron-right" size={16} style={props.styles.icon} />
+                    </TouchableOpacity>
+
+                    <View style={props.styles.divider} />
+
+                    <Button
+                        style={props.styles.bottomButton}
+                        primary
+                        onPress={() => {
+                            props.appSetTcVersion(props.tcAcceptedVersion);
```
Comment (krisboit) [inline src/components/legal/legal.tsx]: i think this is not quite correct, `props.tcAcceptedVersion` are the t&c that were already accepted

File: src/core/constants/app.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export const TOS_VERSION = 1;
+export const TC_VERSION = 1;
```
Comment (krisboit) [inline src/core/constants/app.ts]: i think this is not used anymore

File: src/redux/app/actions.ts [hunk 5]
```diff
@@ -1,11 +1,19 @@
 // actions consts
-export const APP_SET_TOS_VERSION = 'APP_SET_TOS_VERSION';
+export const SET_TC_VERSION = 'SET_TC_VERSION';
+export const SET_ACCEPTED_TC_VERSION = 'SET_ACCEPTED_TC_VERSION';
 
 // actions creators
 
-export const appSetTosVersion = (version: number) => {
+export const appSetTcVersion = (version: number) => {
```
Comment (krisboit) [inline src/redux/app/actions.ts]: why do we need this ?

File: src/redux/app/state.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
 export interface IAppState {
     version: number;
-    tosVersion: number;
+    tcVersion: number;
```
Comment (krisboit) [inline src/redux/app/state.ts]: i think this is not needed...

File: src/screens/connect-hardware-wallet/connect-hardware-wallet.tsx [hunk 7]
```diff
@@ -85,7 +85,7 @@ export class ConnectHardwareWalletScreenComponent extends React.Component<
             connectionActive: false
         };
 
-        if (!props.tosVersion || TOS_VERSION > props.tosVersion) {
+        if (!props.tcVersion || TC_VERSION > props.tcVersion) {
```
Comment (krisboit) [inline src/screens/connect-hardware-wallet/connect-hardware-wallet.tsx]: with the current implementation this is not needed anymore

File: src/screens/create-wallet-mnemonic/create-wallet-mnemonic.tsx [hunk 8]
```diff
@@ -60,7 +60,7 @@ export class CreateWalletMnemonicScreenComponent extends React.Component<
             copied: false
         };
 
-        if (!props.tosVersion || TOS_VERSION > props.tosVersion) {
+        if (!props.tcVersion || TC_VERSION > props.tcVersion) {
```
Comment (krisboit) [inline src/screens/create-wallet-mnemonic/create-wallet-mnemonic.tsx]: same here...

File: src/screens/create-wallet-terms/create-wallet-terms.tsx [hunk 9]
```diff
@@ -11,22 +11,22 @@ import { Icon } from '../../components/icon';
...
     navigation: NavigationStackProp<NavigationState, NavigationParams>;
     styles: ReturnType<typeof stylesProvider>;
 }
 
 export interface IReduxProps {
-    appSetTosVersion: (n: number) => void;
+    appSetTcVersion: typeof appSetTcVersion;
 }
 
 export const CreateWalletTermsScreenComponent = (props: IProps & IReduxProps) => (
```
Comment (krisboit) [inline src/screens/create-wallet-terms/create-wallet-terms.tsx]: this is not cool, we should reuse the content from modal t&c

File: src/components/legal/legal-modal/legal-modal.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+const mapStateToProps = (state: IReduxState) => {
+    return {
+        tcAcceptedVersion: state.app.tcAcceptedVersion
+    };
+};
+
+interface IState {
+    showModal: boolean;
+    tcFirebaseVersion: number;
```
Comment (georgeburduhos) [inline src/components/legal/legal-modal/legal-modal.tsx]: Try to use more general naming like: tcLatestVersion or tcNewVersion

File: src/components/legal/legal.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+import { connect } from 'react-redux';
+import { smartConnect } from '../../core/utils/smart-connect';
+import { appSetAcceptedTcVersion } from '../../redux/app/actions';
+import { INavigationProps } from '../../navigation/with-navigation-params';
+import { TC } from './terms-conditions/terms-conditions';
+import { PrivacyPolicy } from './privacy-policy/privacy-policy';
+
+interface IExternalProps {
+    onAccept: () => void;
+    tcFirebaseVersion: number;
```
Comment (georgeburduhos) [inline src/components/legal/legal.tsx]: same here

File: src/components/legal/privacy-policy/privacy-policy.tsx [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export const PrivacyPolicyComponent = (
+    props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
+) => {
+    return (
+        <View
+            style={[
+                props.styles.container,
+                { paddingTop: Platform.OS === 'ios' && props.showClose ? BASE_DIMENSION * 5 : 0 }
```
Comment (georgeburduhos) [inline src/components/legal/privacy-policy/privacy-policy.tsx]: android and web don't have this problem?

File: src/components/legal/styles.ts [hunk 13]
```diff
@@ -50,13 +43,12 @@ export default (theme: ITheme) =>
...
+        walletTc: {
             fontSize: 17,
             lineHeight: 22,
             textAlign: 'center',
-            color: theme.colors.textSecondary
+            color: theme.colors.textSecondary,
+            marginBottom: BASE_DIMENSION * 10
         },
         text: {
             fontSize: 20,
```
Comment (georgeburduhos) [inline src/components/legal/styles.ts]: try to use theme fonts

File: src/core/utils/remote-feature-config.ts [hunk 14]
```diff
@@ -54,3 +60,12 @@ export const isFeatureActive = (feature: RemoteFeature): boolean => {
...
     return false;
 };
+
+export const getFirebaseTCVersion = (): number => {
+    if (featuresConfig) {
+        const tcVersion = JSON.parse(featuresConfig[RemoteFeature.TC_VERSION]);
+        return tcVersion;
+    }
+
+    return;
```
Comment (georgeburduhos) [inline src/core/utils/remote-feature-config.ts]: what happens if there is no featureConfig - Firebase call crashes

File: src/components/legal/privacy-policy/privacy-policy.tsx (hunk 12 above)
Comment (NovacAndrei) [reply src/components/legal/privacy-policy/privacy-policy.tsx]: The LegalModal it's not shown on the web Platform.
On Android it's working, but it has to be tested and verified on a device with notch.

On iOS we have the following issue:

image

**Pull request merged**